pkg/web/apps/apps_service.go

```diff
@@ -108,23 +121,37 @@ func (a *appsService) UnbindingAppByAppID(appID string) error {
 	return nil
 }
 
-func (a *appsService) BindingAppByApp(appId, name string) error {
+func (a *appsService) CreateUpdateApp(app models.App) error {
 
 	// Check if it exist
-	app, err := a.repository.GetAppByAppID(appId)
+	appStored, err := a.repository.GetAppByAppID(app.AppID)
 
 	// If it is new then create an app
 	if err != nil && err == models.ErrNotFound {
 		id := helpers.GetUUID()
-		return a.repository.CreateApp(id, appId, name)
+		return a.repository.CreateApp(id, app.AppID, app.AppName)
 	}
 
+	// return error in the creation
 	if err != nil {
 		return err
 	}
 
-	// if is deleted so just reactive the existent app
-	return a.repository.UnDeleteAppByAppID(app.AppID)
+	// check if is disabled
+	if appStored.DeletedAt != "" {
+		// if is deleted so just reactive the existent app
+		if err := a.repository.UnDeleteAppByAppID(app.AppID); err != nil {
+			return err
+		}
+	}
+
+	// update the name if it was changed
+	if app.AppName != "" && appStored.AppName != app.AppName {
```

Comment:
We should not be updating the appName on a /apps POST. If a user wants to update a field within a specific resource it would be a PATCH on `/apps/:id`.

Reply:
OK. But the disabled_at is also updated here. Should we create an endpoint for update the name only using the PATCH method or since the deleted_at update is done here we can move with it in the post as well?

Reply:
The deleted_at is an internal only field, it is never sent from an end user to be updated. It captures the soft deletion of this entity, this is why we are ok to update the internal deleted_at field on this POST to undo the soft deletion.

An update to an individual resource from an end user should be done through a PUT or PATCH of the individual resource rather than through a filter on the resources collection.

Reply:
Question for @wei-lee ^ Will be possible change the appName from MDC?

Reply:
Hi @davidffrench,

I don't know but I think that yes. So, I will move this for a PACTH if is possible  change the name by MDC. @wei-lee could you let us know if users will or not be able to change the name of the app by the MDC?

Reply:
yes, it's possible in MDC. however, I don't think we are going to update the app names across all services at least for now. It is a bit tricky to do and we probably need to think about it a bit more.